### PR TITLE
added problem4 - takeFirst

### PR DESCRIPTION
**Pull request by marcinwysocki**

### CODE REVIEW

File: tests/problem4.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    });
+
+    it('resolves with an empty array if no promises have been passed', async () => {
+        expect(await takeFirst(3)).toEqual([]);
+    });
+
+    it('resolves with given number of first resolved promises', async () => {
+        const promise1 = setTimeoutPromise(500);
+        const promise2 = setTimeoutPromise(700);
+        const promise3 = setTimeoutPromise(800);
```
Comment (mpodlasin) [inline tests/problem4.test.js]: timeouts seem to long to me - won't tests take too long? I would stick to timeouts < 100ms

File: tests/problem4.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        expect(
+            await takeFirst(2, promise1(), promise3(), promise4(), promise2()),
+        ).toEqual([500, 700]);
+        expect(
+            await takeFirst(3, promise1(), promise3(), promise4(), promise2()),
+        ).toEqual([500, 700, 800]);
+    });
+
+    it('rejects if one of the promises rejects before given number of them has been resolved', async () => {
```
Comment (mpodlasin) [inline tests/problem4.test.js]: how about case where we expected 3 promises to resolves, and 3 has been resolved, but after that fourth did throw?

File: tests/problem4.test.js (hunk 1 above)
Comment (marcinwysocki) [reply tests/problem4.test.js]: Yeah, I've shortened them.

File: tests/problem4.test.js (hunk 2 above)
Comment (marcinwysocki) [reply tests/problem4.test.js]: I was convinced I had written a test for this... Now I have ;)

**Pull request merged**